confluent_kafka/avro/__init__.py

```diff
@@ -33,7 +33,7 @@ def __init__(self, config, default_key_schema=None,
                    for key, value in config.items() if key.startswith("schema.registry")}
 
         if sr_conf.get("basic.auth.credentials.source") == 'SASL_INHERIT':
-            sr_conf['sasl.mechanisms'] = config.get('sasl.mechanisms', '')
+            sr_conf['sasl.mechanism'] = config.get('sasl.mechanism', '')
```

Comment:
sasl.mechanism is just an alias for sasl.mechanisms, so this change will break existing applications that use sasl.mechanisms.

What we'll need to do here is something like:
`sr_conf['sasl.mechanism'] = config.get('sasl.mechanism', config.get('sasl.mechanisms', ''))`